qcengine/tests/test_config.py

```diff
@@ -59,6 +60,31 @@ def test_node_skip_environ():
     assert node.scratch_directory is None
 
 
+@pytest.mark.parametrize(
+    "envvars,scr,ans",
+    [
+        ({"SCRATCH": "/my_scratch"}, "$SCRATCH", "/my_scratch"),
+        ({"SCRATCH": "/my_scratch"}, "${SCRATCH}", "/my_scratch"),
+        # ({"SCRATCH": "/my_scratch"}, "%SCRATCH%", "/my_scratch"),  # only valid on Windows
+        (
+            {"PBS_DIR": "/cluster/scr/job", "JOBNUM": "12345"},
+            "$PBS_DIR/psi_scratch/$JOBNUM",
+            "/cluster/scr/job/psi_scratch/12345",
+        ),
+        ({"QCSCR": "qcscr", "USER": "johndoe"}, "/scratch/${USER}/$QCSCR", "/scratch/johndoe/qcscr"),
+        ({}, "~", f"{os.environ.get('HOME')}"),
+        ({"QCSCR": "qcscr"}, "~/scratch/$QCSCR", f"{os.environ.get('HOME')}/scratch/qcscr"),
+        ({}, "$HOME", f"{os.environ.get('HOME')}"),
+        # ({}, "$RANDOM_NOVAR", "$RANDOM_NOVAR"),  # new behavior?
+        ({}, "$RANDOM_NOVAR", None),  # longstanding behavior
```

Comment:
Ready for review, @bennybp . Besides expanding envvars, `parse_environment` was also nulling values that looked like undefined envvars (start with `$`). That was tested behavior, so I put it back. Basically a question of do we want the "new behavior" or the "longstanding behavior".

Reply:
I think the longstanding behavior is iffy, but I guess I'm hesitant to change it. Other than that, looks good to me

Reply:
Agreed. I didn't expect the longstanding behavior, but I don't fancy changing it for a patch release.